layers/cars.py

```diff
@@ -0,0 +1,153 @@
+    """
+    try:
+        from arcgis.gis import GIS
+    except ImportError:
+        print("Unable to import `arcgis`. Please install the `arcgis` package, using the command `pip install requirements-non-foss.txt.")
+        return
+    
+    # Anonymous access to ArcGIS Online
+    gis = GIS()
+
+    # Get the `Item`, then, `FeatureLayer` then 'FeatureSet`:
+    agol_item = gis.content.get(serviceItemId)
+    print(f"Got item: {agol_item}")
+    print(f"item metadata: {agol_item.metadata}")
+
+    agol_layer = agol_item.layers[0]
+
+    # Get the last edit datetime for the layer
+    # print(f"Got layer: {agol_layer.properties}")
+    lyr_props = agol_layer.properties
+    # Epoch time in milliseconds - convert to datetime
+    lyr_last_edit = lyr_props.get("editingInfo", {}).get("lastEditDate", None)
+    if lyr_last_edit:
+        lyr_last_edit = datetime.datetime.fromtimestamp(lyr_last_edit/1000)
+
+    print(f"last_edit: {lyr_last_edit}")
+
+    # If the output file exists, check the last edit time
+    output_last_edit = _last_update(output_file)
+
+    if not force and output_last_edit and lyr_last_edit and output_last_edit > lyr_last_edit:
+        print(f"Output file is up-to-date: {output_file}")
+        return
+    
+    print(f"Output file is out-of-date: {output_file}")
+
+    agol_feature_set = agol_layer.query()
+    print(f"Got feature set: {len(agol_feature_set)}")
+    
+    # Write to geojson file
+    with open(output_file, "w") as f:
+        f.write(agol_feature_set.to_geojson)
+
+    print("Done")
+
+
+#Code from https://github.com/dabreegster/popgetter/tree/main
+def download_vehicle_ownership(census_url, working_dir):
+    print("Retrieving Vehicle Ownership Census Data")
+    # TODO This doesn't give programmatic access to the name of the downloaded file. This is a problem if/when the url's slug contains many parameters (where the parameter order is not guaranteed or if the parameter string is not a valid filename for all OSes) 
+    result = subprocess.check_call(["wget", "-N", census_url], cwd=working_dir)
+    print(f"result = {result}")
+
+
+#%% Get Input Data
+
+WORKING_DIR = "data"
+
+#Get OA geometries
+output_areas_serviceItemID = "6c6743e1e4b444f6afcab9d9588f5d8f"
+output_areas_geojson_path = f"{WORKING_DIR}/oa_from_agol.geojson"
+download_from_arcgis_online(output_areas_serviceItemID, output_areas_geojson_path, force=False)
+
+#Get Car Ownership Census Data
+
+#Below code not working - cannot find file (althugh URL work in a web browser)
+#To work around download file from below url and save in WORKING_DIR
+
+# CENSUS_URL = "https://static.ons.gov.uk/datasets/a20437fb-ae7f-439b-bc91-de261335038b/TS045-2021-3-filtered-2023-03-13T16:49:47Z.csv"
+# download_vehicle_ownership(CENSUS_URL, WORKING_DIR)
+
+carOwnerFile = 'TS045-2021-3-filtered-2023-03-13T16 49 47Z.csv'
+
+#%% Read data in
+
+#OAs shapefile as geopandas dataframe
+oas = gpd.read_file(output_areas_geojson_path)
```

Comment:
Can we transform to WGS84 coordinates (aka EPSG:4326) too?

Reply:
CRS already WGS84, added print statement to show this.